### ISSUE TITLE
Is it possible to add zip_extract method ?

### ISSUE THREAD

**Issue by Liloupar** [perron2/async-zip#5]
[extern ZIP_EXPORT int zip_extract(const char *zipname, const char *dir,](https://github.com/perron2/async-zip/blob/f17ee33a38bf970429a835ed1d237912b1a3a9bd/ios/Classes/zip.h#LL431C28-L431C28)

it is in zip.h. 
thank you.

**Comment by claudiofelber** [comment]
zip_extract() is directly operating on the file system paths which probably is only of limited use on Android and iOS due to the sandboxed file system. Could you please briefly outline a typical use case? This would help in designing an appropriate API.

**Comment by Liloupar** [comment]
I am now using the `extractFileToDisk` method in this package [archive](https://pub.dev/packages/archive) to decompress the epub file to the specified directory. 
But its performance and compatibility are far behind this c library.
So I was expecting to do the same with zip_extract.

**Comment by claudiofelber** [comment]
I have published version 0.1.0 of this package. It contains two new functions `extractZipArchiveSync` and `extractZipArchive` which allow extracting an entire Zip archive to a folder. An optional callback can be specified in order to receive information about the extraction progress.

**Comment by Liloupar** [comment]
> I have published version 0.1.0 of this package. It contains two new functions `extractZipArchiveSync` and `extractZipArchive` which allow extracting an entire Zip archive to a folder. An optional callback can be specified in order to receive information about the extraction progress.

Awesome
I'm going to try it today

**Comment by Liloupar** [comment]
I gave it a try, and it worked really well.
Thank you so much! However, I noticed one thing: the extractZipArchive method didn't return the int value returned by the c method, which indicates whether the extraction was successful.

>Returning a negative value from the callback will cause abort and return an error.
https://github.com/perron2/async-zip/blob/f17ee33a38bf970429a835ed1d237912b1a3a9bd/ios/Classes/zip.h#LL420C4-L421C10

**Comment by claudiofelber** [comment]
Yes, that's true. I am currently using the C function only in the case where extractZipArchive is used without a callback function. In case a callback function is specified, the unpacking of the archive is done with Dart code (based on other C library function of course). This is because native C callbacks currently do not work properly across Dart isolates, which are used to provide the asynchronous functionality.

For my own implementation of the ZIP archive extraction I could not yet decide on a reasonable return value. There are various possibilities:
- Return the first error detected
- Return the last error detected
- Return a list of all errors

There is also the question of what should happen in the event of an error:
- Should the extraction process abort?
- Or should the extraction continue and try to extract as many files as possible?

The answers to the aforementioned questions are likely use case specific. Therefore at least the behavior in the event of an error (abort or continue) should probably be configurable. What are your thoughts about all of this?

**Comment by Liloupar** [comment]
I think all of these options are viable, but I lean towards the first one.

> Return the first error detected
> Should the extraction process abort
